apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V114__s3_file_tracking.sql

```diff
@@ -0,0 +1,50 @@
+
+/*
+ * Table to track CCW S3 manifest files in database.
+ */
+CREATE TABLE s3_manifest_files (
+    manifest_id bigint NOT NULL,
+    s3_key varchar(1024) NOT NULL,
+    status varchar(24) NOT NULL,
+    status_timestamp timestamp with time zone,
+    manifest_timestamp timestamp with time zone NOT NULL,
+    discovery_timestamp timestamp with time zone NOT NULL,
+    CONSTRAINT pk_s3_manifest_files PRIMARY KEY (manifest_id)
+);
+
+/*
+ * Sequence used to generate primary key manifest_id
+ */
+create sequence s3_manifest_files_manifest_id_seq ${logic.sequence-start} 1 ${logic.sequence-increment} 1;
+
+/*
+ * Index used when checking status of a file from S3 bucket
+ */
+create unique index idx_s3_manifest_files_s3_key on s3_manifest_files(s3_key);
+
+/*
+ * Table to track CCW S3 data files in database.
+ */
+CREATE TABLE s3_data_files (
+    manifest_id bigint NOT NULL,
+    index smallint NOT NULL,
+    file_name varchar(128) NOT NULL,
+    file_type varchar(50) NOT NULL,
+    s3_key varchar(1024) NOT NULL,
```

Comment:
Probably not but one of the nice things about postgresql is that all varchars are the same size in the db.  The limit just applies to maximum it will let you store.  So setting the max very high does not make the table any larger.